checker/src/main/java/org/checkerframework/checker/mustcall/CreatesObligationElementSupplier.java

```diff
@@ -0,0 +1,33 @@
+package org.checkerframework.checker.mustcall;
+
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/**
+ * This interface should be implemented by all type factories that can provide the ExecutableElement
+ * needed to call {@link AnnotationUtils#getElementValueArray} when {@link
+ * MustCallTransfer#getCreatesObligationExpressions(MethodInvocationNode,
+ * GenericAnnotatedTypeFactory, CreatesObligationElementSupplier)} is called. This interface is
+ * needed so any type factory with these elements can be used to call that method, not just the
+ * MustCallAnnotatedTypeFactory (in particular, the consistency checker needs to be able to call
```

Comment:
Why does it need that?  Couldn't it have a field that contains the given elements?  Then it would use its own copy rather than asking a type factory for a copy.

Reply:
The elements can only be computed by a type factory, because doing so requires a `ProcessingEnvironment`, which is only available as a field of the type factory. Because of this, I've been operating on the assumption that a given element should only be used by the type factory used to create it; I haven't tested what would happen if I tried to share these between multiple factories (but if it was independent, then why would it need a field of the factory to be constructed?). This interface exists so that a type factory can declare that it will provide the relevant elements.